src/aliases/store/postgres/alias.go

```diff
@@ -23,10 +23,13 @@ func NewAlias(pgClient postgres.Client) *AliasStore {
 }
 
 func (s *AliasStore) CreateAlias(ctx context.Context, registry aliasent.RegistryName, alias aliasent.Alias) (*aliasent.Alias, error) {
-	a := aliasmodels.AliasFromEntity(alias)
+	a, err := aliasmodels.AliasFromEntity(alias)
+	if err != nil {
+		return nil, err
+	}
 	a.RegistryName = aliasmodels.RegistryName(registry)
 
-	err := s.pgClient.Insert(ctx, &a)
+	err = s.pgClient.Insert(ctx, &a)
```

Comment:
You should be logging here and returning a specific error message. Same comment for all functions of this Agent/Repository. Otherwise you are returning specific postgres errors to the users.

Reply:
For me logging is already dealing with the error. Logging + returning an error could be doing the work twice.
We need to add that to the list :)